content/src/content/jcr_root/apps/core/wcm/sandbox/commons/ui/clientlibs/panelcontainer/js/PanelContainer.js

```diff
@@ -62,19 +63,30 @@
          * [Content Frame]{@link Granite.author.ContentFrame} and lets the related UI widget handle the operation.
          *
          * @param {Number} index Index of the panel to navigate to
+         * @fires CQ.CoreComponents.PanelContainer#cmp-panelcontainer-navigated
          */
         navigate: function(index) {
             if (this._config.panelContainerType) {
                 // assign a path-based ID data attribute to the panel container HTMLElement
                 // for correct forwarding of operations in the content frame message subscription.
                 this._config.el.dataset["cmpPanelcontainerId"] = this._config.path;
 
-                Granite.author.ContentFrame.postMessage(MESSAGE_ID, {
+                var data = {
                     id: this._config.path,
                     type: this._config.panelContainerType.name,
                     operation: "navigate",
                     index: index
-                });
+                };
+
+                Granite.author.ContentFrame.postMessage(MESSAGE_ID, data);
+
+                setTimeout(function() {
+                    channel.trigger({
+                        type: "cmp-panelcontainer-navigated",
+                        id: data.id,
+                        index: data.index
+                    });
+                }, NAVIGATE_DELAY);
```

Comment:
@vladbailescu, @jckautzmann - this delay is a little brittle, it's an approximation on the time to update/navigate the UI component after posting the message to the content frame. Without receiving a return message I don't see a better approach unless you have an idea?

Reply:
@richardhand As you pointed, we could trigger a `navigationDone` event in the Carousel/Tabs js logic when the navigation is done and bind a handler here to listen to this event. I don't see other options.